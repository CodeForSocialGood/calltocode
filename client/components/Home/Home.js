import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import Checkbox from 'material-ui/Checkbox'
import { FormGroup, FormLabel, FormControlLabel } from 'material-ui/Form'

import ProjectActionCreator from '../../actions/project'
import ListOfProjects from '../ListOfProjects/ListOfProjects'
import styles from './Home.scss'
import { causes, technologies } from '../shared/constants'

class Home extends Component {
  constructor (props) {
    super(props)
    this.handleCheckbox = this.handleCheckbox.bind(this)
    this.renderList = this.renderList.bind(this)
    this.state = {
      filters: {}
    }
  }

  componentDidMount () {
    this.props.onLoad()
  }

  handleCheckbox (event, checked) {
    console.log(event.target.name, event.target.value, checked)
  }

  renderList (items, type) {
    return items.map((item, index) => {
      return (
        <FormControlLabel key={index}
          control={ <Checkbox name={type} value={item} onChange={this.handleCheckbox} /> }
          label={item}
          className={styles.checkBoxes}
        />
      )
    })
  }

  render () {
    return (
      <div className={styles.homeGrid}>
        <section className={styles.filterSection}>
          <span className={styles.filterTitle}><h1>{'Filters'}</h1></span>
          <FormLabel className={styles.causeTitle}>Causes</FormLabel>
          <FormGroup column className={styles.causesList}>
            { this.renderList(causes, 'causes') }
          </FormGroup>
          <FormLabel className={styles.techTitle}>Tech</FormLabel>
          <FormGroup column className={styles.techList}>
            { this.renderList(technologies, 'technologies') }
          </FormGroup>
        </section>

        <ListOfProjects
          title={'Click To Apply'}
          projects={this.props.projects}
          className={styles.projectList}
        />
      </div>
    )
  }
}

function mapStateToProps (state) {
  return {
    projects: state.project.projects
  }
}

const mapDispatchToProps = {
  onLoad: ProjectActionCreator.fetchAllProjects
}

Home.propTypes = {
  onLoad: PropTypes.func,
  projects: PropTypes.array
}

export default connect(mapStateToProps, mapDispatchToProps)(Home)
