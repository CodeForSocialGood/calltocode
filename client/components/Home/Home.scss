@import '../../scss/partials/partials';

.title {
	font-size: 20px;
	margin: 0 0 15px -240px;
	padding: 15px;
	padding-left: 100px;
	color: white;
	width: 92vw;
	background-color: $backgound-orange;
}

.homeGrid {
	background-color: $background-gray;
	display: grid;
	margin-top: 30px;
	margin-left: -150px;
	grid-template-columns: auto 85%;
	grid-template-areas:
		"filterSection projectList";
}

.projectList {
	grid-area: projectList;
	display: grid;
	grid-row-gap: 20px;
	padding: 0 30px;
	height: auto;
}

.filterSection {
	padding: 0 30px 10px;
	grid-area: filterSection;
	box-shadow: 0 4px 80px 0 rgba(0, 0, 0, 0.1);
	display: grid;
	grid-template-rows: 60px 30px auto 30px auto;
	grid-template-columns: 300px;
	grid-template-areas:
		"filterTitle"
		"causeTitle"
		"causes"
		"techTitle"
		"technologies";
	grid-row-gap: 10px;
	height: 800px;
}

.filterTitle {
	grid-area: filterTitle;
}

.causeTitle {
	grid-area: causeTitle;
	border-radius: $filterTitleRadius;
	color: $listColor;
	font-weight: bold;
	height: 30px;
	display: flex;
	align-items: center;
	justify-content: center;
	background-color: $filterTitleColor;
}

.techTitle {
	grid-area: techTitle;
	border-radius: $filterTitleRadius;
	color: $listColor;
	font-weight: bold;
	height: 30px;
	display: flex;
	align-items: center;
	justify-content: center;
	background-color: $filterTitleColor;
}

.causesList {
	grid-area: causes;
}

.techList {
	grid-area: technologies;
}

.checkBoxes {
	margin-top: -2px;
}
