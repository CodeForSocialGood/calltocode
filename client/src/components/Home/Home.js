import React, { Component } from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'

import ProjectActionCreator from '../../actions/project'
import ListOfProjects from '../ListOfProjects/ListOfProjects'

class Home extends Component {
  componentDidMount () {
    this.props.populateOpps()
  }

  render () {
    return (
      <ListOfProjects
        title={'Click To Apply'}
        projects={this.props.projects} />
    )
  }
}

function mapStateToProps (state) {
  return {
    projects: state.project.projects
  }
}

const mapDispatchToProps = {
  populateOpps: ProjectActionCreator.populateOpps
}

Home.propTypes = {
  projects: PropTypes.array,
  populateOpps: PropTypes.func
}

export default connect(mapStateToProps, mapDispatchToProps)(Home)
