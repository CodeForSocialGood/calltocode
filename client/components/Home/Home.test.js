/* eslint-env jest */
import React from 'react'
import {shallow} from 'enzyme'

import {Home} from './Home'
let mockProps

beforeEach(() => {
  mockProps = {
    onLoad: () => {},
    projects: []
  }
})

describe('<Home />', () => {
  describe('Projects is empty', () => {
    test('component renders correctly', () => {
      const component = shallow(<Home {...mockProps}/>)
      expect(component).toMatchSnapshot()
    })
  })
  describe('Projects is not empty', () => {
    beforeEach(() => { mockProps.projects = [{}] })

    test('component renders correctly', () => {
      const component = shallow(<Home {...mockProps}/>)
      expect(component).toMatchSnapshot()
    })
  })
})
