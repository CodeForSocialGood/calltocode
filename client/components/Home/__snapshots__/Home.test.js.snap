// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> Projects is empty component renders correctly 1`] = `
<Connect(ListOfProjects)
  projects={Array []}
  title="Click To Apply"
/>
`;

exports[`<Home /> Projects is not empty component renders correctly 1`] = `
<Connect(ListOfProjects)
  projects={
    Array [
      Object {},
    ]
  }
  title="Click To Apply"
/>
`;
